testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +113,76 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
+
+    def test_check_set_unit_density_none(self, D):
+        del D.units['density']
+        D._check_set_unit({})
+        assert D.units['density'] == None
+        assert D._check_set_unit(D.units) == None
```

Comment:
Which bit of the code is being tested by passing the same units to itself?

Also - same issue as previous comment, unless I'm missing something obvious `_check_set_unit` has no return so you need to assert on `D.units`.

Reply:
Yes, sorry about that duplicated `test_check_set_unit_density_none`, this one was supposed to be deleted. The unit test is intended to cover lines 772-773 in `density.py`.